src-electron/generator/helper-zcl.js

```diff
@@ -2209,6 +2209,78 @@ async function if_mfg_specific_cluster(clusterId, options) {
   }
 }
 
+/**
+ *
+ * @param {*} value
+ * @param {*} size
+ * @returns The negative hex value for a given positive hex value padded with
+ * the given size.
+ */
+async function convert_float_to_hex(value, size) {
+  let res = ''
+  const arrayBuffer = new ArrayBuffer(size)
+  const dataView = new DataView(arrayBuffer)
+  dataView.setFloat32(0, value, false)
+  for (let i = 0; i < size; i++) {
+    let t = dataView.getUint8(i).toString(16)
+    res += t == 0 ? '00' : t
+  }
+  return '0x' + res
+}
+
+/**
+ *
+ * @param {*} value
+ * @param {*} size
+ * @returns the negative hex value for a given positive hex value padded with
+ * the given size.
+ */
+async function get_negative_hex_value(value, size) {
+  let res = ''
+  // Performing negation using the binary form
+  let tempBinary = ''
+  let byteArray = value.replace('0x', '').match(/.{1,2}/g)
+  for (let b of byteArray) {
+    let binValue = parseInt(b, 16).toString(2)
+    tempBinary += '0'.repeat(8 - binValue.length) + binValue
+  }
+  let paddingSize = 8 * size - byteArray.length * 8
+  tempBinary = '0'.repeat(paddingSize) + tempBinary
+  tempBinary = tempBinary.replaceAll('1', 't')
+  tempBinary = tempBinary.replaceAll('0', '1')
+  tempBinary = tempBinary.replaceAll('t', '0')
+
+  // Adding 1 to the binary number
+  if (tempBinary.charAt(tempBinary.length - 1) == '0') {
+    tempBinary = tempBinary.replace(/0$/, '1')
+  } else {
+    let carry = '1'
+    let result = ''
+    for (let i = tempBinary.length - 1; i >= 0; i--) {
+      if (tempBinary.charAt(i) == '1') {
+        if (carry == '1') {
+          result = '0' + result
+        } else {
+          result = tempBinary.charAt(i) + result
+        }
+      } else {
+        if (carry == '1') {
+          result = '1' + result
+          carry = '0'
+        } else {
+          result = tempBinary.charAt(i) + result
+        }
+      }
+    }
+    tempBinary = result
```

Comment:
Wow. Is there seriously no package out there that does this? This seems very complicated...